Update handleCardSetup()

geekygrappler opened the pull request:
- Stripe deprecated handleCardSetup() https://stripe.com/docs/stripe-js/reference#deprecated
- Looks like you now need to use confirmCardSetup() for the same thing https://stripe.com/docs/stripe-js/reference#stripe-confirm-card-setup

There probably are more depreciations, but this is the one that I was using, so I noticed it had broken.

I just did a find a replace, not overthinking, but it works in my app and the tests pass 💆‍♂️

lindyhopchris commented:
Thanks for this. I think because it is deprecated but not removed, we should leave `handleCardSetup` in this package, and *add in* `confirmCardSetup`. Otherwise this is quite a big breaking change, which isn't strictly necessary when the deprecated method does still exist.

geekygrappler commented:
Yeah you're right I'll change it to add in this method and leave the old one.

geekygrappler commented:
Haha I actually had a typo which made me think `handleCardSetup()` wasn't working anymore, which is not the case. So as my penance I will do this PR properly.

geekygrappler committed:
add confirmCardSetup() method

- Stripe deprecated handleCardSetup() https://stripe.com/docs/stripe-js/reference#deprecated
- Looks like you now need to use confirmCardSetup() for the same thing https://stripe.com/docs/stripe-js/reference#stripe-confirm-card-setup

geekygrappler commented:
🚀

geekygrappler commented:
hmmm tests pass locally, I think it's one version failing, can admin retry the CI please?

lindyhopchris commented on addon/utils/stripe-mock.js:
```diff
@@ -23,7 +23,7 @@ StripeMock.prototype.retrievePaymentIntent = function() {};
 StripeMock.prototype.handleCardAction = function() {};
 StripeMock.prototype.confirmPaymentIntent = function() {};
-StripeMock.prototype.handleCardSetup = function() {};
+StripeMock.prototype.confirmCardSetup = function() {};
```
I think the test is failing because you've removed `handleCardSetup` here - can you put it back in?

geekygrappler replied:
Ooops, sorry I missed that 🙏 . Weird that the other versions of ember passed...

lindyhopchris commented:
Thanks!
